cli/ops/net.rs

```diff
@@ -129,6 +221,59 @@ struct ListenArgs {
   port: u16,
 }
 
+#[allow(dead_code)]
+struct TcpListenerResource {
+  listener: tokio::net::TcpListener,
+  task: Option<futures::task::Task>,
+  local_addr: SocketAddr,
+}
+
+impl CoreResource for TcpListenerResource {}
+
+impl Drop for TcpListenerResource {
+  fn drop(&mut self) {
+    self.notify_task();
+  }
```

Comment:
👍

Reply:
I dont fully understand why we need this.. @bartlomieju can you please explain why we need to explicity wake?

Reply:
@95th listener can be closed when there is on-going accept going. Storing task that handles that accept and notifying it allows accept promise to finish instead of hanging forever.